src/gadget/is_equal_rot.rs

```diff
@@ -0,0 +1,296 @@
+use crate::gadget::Variable;
+use halo2::{
+    circuit::{Chip, Region},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Selector, VirtualCells},
+    poly::Rotation,
+};
+use pasta_curves::arithmetic::FieldExt;
+use std::marker::PhantomData;
+
+pub(crate) trait IsEqualRotInstruction<F: FieldExt, const R: i32> {
+    /// Given an input pair `(value.rot(R), value)`, helps assign `value` and
+    /// `(value - value.rot(R)).invert()`, returns variable `value`
```

Comment:
```suggestion
    /// Given an existing variable at `value_rot` at rotation R, and a `value` to be witnessed:
    ///    - witnesses `value`;
    ///    - witnesses `inv0(value - value_rot)`, where `inv0(X)` is 0 when `X` = 0,
    ///      and `1/X` otherwise;
    ///    - and returns the witnessed variable `value`.
```